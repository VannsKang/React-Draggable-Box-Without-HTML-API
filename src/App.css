.outerContainer {
	position: absolute;
	display: flex;
	justify-content: center;
	align-items: center;
	width: 100%;
	height: 100%;
}

.container {
	display: flex;
	justify-content: center;
	align-items: center;
	width: 800px;
	height: 500px;
	margin: auto;
	background-color: #333;
	border-radius: 10px;
	overflow: hidden;
	touch-action: none;
}

.draggableBox {
	position: absolute;
	width: 150px;
	background-color: #ca6f1e;
	border-radius: 5px;
	user-select: none;
	touch-action: none;
}

.draggableBox:active {
	background-color: #f9e79f;
}

.draggableBox:hover {
	cursor: pointer;
}

.box {
	display: flex;
	justify-content: center;
	align-items: center;
}
