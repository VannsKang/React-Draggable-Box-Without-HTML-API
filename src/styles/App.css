.draggable-title {
	position: absolute;
	top: 4rem;
	left: 50%;
	transform: translateX(-50%);
	z-index: 10;
}

.draggable-title div {
	font-family: 'Vibur', cursive;
	text-align: center;
}

.outerContainer {
	position: absolute;
	display: flex;
	justify-content: center;
	align-items: center;
	width: 100%;
	height: 100%;
	padding-top: 10rem;
}

.container {
	display: flex;
	justify-content: center;
	align-items: center;
	height: 500px;
	margin: auto;
	background-color: #333;
	border-radius: 10px;
	overflow: hidden;
	touch-action: none;
}

.draggableBox {
	position: absolute;
	background-color: #333;
	user-select: none;
	padding: 15px 0;
	touch-action: none;
}

.draggableBox:active {
	background: rgb(63, 94, 251);
	background: radial-gradient(circle, rgba(63, 94, 251, 1) 0%, rgba(252, 70, 107, 1) 100%);
}

.draggableBox:hover {
	cursor: pointer;
}

.box {
	display: flex;
	justify-content: center;
	align-items: center;
	font-weight: 900;
}

/* neon style */

.shadow {
	--tw-shadow: 0 20px 25px -5px rgba(0, 0, 0, 0.1), 0 10px 10px -5px rgba(0, 0, 0, 0.04);
	box-shadow: 0 35px 60px -15px rgba(0, 0, 0, 0.3);
}

.gradient {
	background: rgb(63, 94, 251);
	background: linear-gradient(90deg, rgba(63, 94, 251, 1) 0%, rgba(252, 70, 107, 1) 100%);
}

.neon-border {
	border: 3px solid #fff;
	border-radius: 0.8rem;

	box-shadow: 0 0 0.2rem #fff, 0 0 0.2rem #fff, 0 0 2rem #bc13fe, 0 0 0.8rem #bc13fe,
		0 0 2.8rem #bc13fe, inset 0 0 1.3rem #bc13fe;
}

.neon-title {
	font-family: 'Vibur', cursive;
	color: #fff;
	text-shadow: 0 0 7px rgb(63, 94, 251), 0 0 10px rgb(63, 94, 251), 0 0 21px rgb(63, 94, 251),
		0 0 42px rgb(63, 94, 251);
	animation: flicker 2s infinite alternate;
}

@keyframes flicker {
	0%,
	18%,
	22%,
	25%,
	53%,
	57%,
	100% {
		text-shadow: 0 0 4px rgb(63, 94, 251), 0 0 11px rgb(63, 94, 251), 0 0 19px rgb(63, 94, 251),
			0 0 40px rgb(63, 94, 251);
	}

	20%,
	24%,
	55% {
		text-shadow: none;
	}
}

/* media query */

/* mobile */
@media (max-width: 1023px) {
	.draggable-title div {
		font-size: 1.5rem;
		line-height: 1.3;
	}

	.container {
		width: 300px;
	}

	.draggableBox {
		width: 80px;
		padding: 10px 0;
	}

	.box {
		font-size: 1.2rem;
	}
}

/* desktop */
@media (min-width: 1024px) {
	.draggable-title div {
		font-size: 3.5rem;
		line-height: 1.3;
		white-space: nowrap;
	}

	.container {
		width: 800px;
	}

	.draggableBox {
		width: 150px;
		padding: 15px 0;
	}

	.box {
		font-size: 2rem;
	}
}
