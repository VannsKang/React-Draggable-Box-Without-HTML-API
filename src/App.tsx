import Box from '@/components/Box';
import Draggable from '@/components/Draggable';

import './App.css';

function App() {
	return (
		<div className='App' data-testid='app'>
			<Draggable>
				<Box />
			</Draggable>
		</div>
	);
}

export default App;
