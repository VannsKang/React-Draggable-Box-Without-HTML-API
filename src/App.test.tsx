import { render, screen, cleanup } from '@testing-library/react';
import App from './App';

afterEach(cleanup);

test('renders App check', () => {
	render(<App />);
	const app = screen.getByTestId('app');
	expect(app).toBeInTheDocument();
});
