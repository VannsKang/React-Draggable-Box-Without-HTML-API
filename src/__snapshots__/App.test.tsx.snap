// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="outerContainer"
    >
      <div
        class="container"
      >
        <div
          class="draggableBox"
        >
          <h1
            class="box"
          >
            BOX
          </h1>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
